src/DayPickerInput.js

```diff
@@ -312,7 +313,9 @@ export default class DayPickerInput extends React.Component {
     if (this.state.showOverlay === false) {
       return;
     }
-    this.setState({ showOverlay: false });
+    this.setState({ showOverlay: false }, () => {
+      if (this.props.onClose) this.props.onClose();
```

Comment:
I wonder if this is the place where we should call the event handler. I believe we should put it in a `componentDidUpdate`:

```jsx
componentDidUpdate(prevProps, prevState) {
    if (this.props.onDayPickerHide && this.state.showOverlay && !prevState.showOverlay) {
        this.props.onDayPickerHide();
    }
}
```

What do you think?

Reply:
Not sure about this, it seems to overlook a simple solution. We want it to fire just after the DayPicker is hid and that's when the callback is fired. While it would also do that in `componentDidUpdate` from a performance point of view it means that every time the component is updated it would check the if statement - which I do not think is necessary. 
What are your thoughts?

Reply:
I share with you the concerns about performance (which are more ideal, though). The reason I was thinking about this change, is that we must be sure the overlay is _always_ hidden only by calling `hideDayPicker` method, and not just by setting the state (which won't call the event handler). So in conclusion, we can go with this. We should just take extra care and hide the overlay only by calling that method.